### ISSUE TITLE
locate raise an exception when used on a functor

### ISSUE THREAD

**Issue by Khady** [ocaml/merlin#845]
If locate is called on `Make` in the given code, an exception is raised. The trace sent to emacs is pasted bellow.

merlin 3.1.0
ocaml 4.05.0

```ocaml
module SM = Map.Make(String)
```

```
cond: unknown answer: "exception":"File \"src/analysis/typedtrie.ml\", line 513, characters 16-22: Assertion failed
Raised at file \"src/analysis/typedtrie.ml\", line 513, characters 16-28
Called from file \"src/analysis/typedtrie.ml\", line 552, characters 8-31
Called from file \"src/analysis/locate.ml\", line 388, characters 8-68
Called from file \"src/analysis/locate.ml\", line 703, characters 10-57
Called from file \"src/analysis/locate.ml\", line 837, characters 6-65
Called from file \"src/utils/std.ml\", line 638, characters 8-12
Re-raised at file \"src/utils/std.ml\", line 640, characters 30-39
Called from file \"src/frontend/query_commands.ml\", line 498, characters 6-119
Called from file \"src/utils/local_store.ml\", line 29, characters 8-12
Re-raised at file \"src/utils/local_store.ml\", line 37, characters 4-15
Called from file \"src/kernel/mocaml.ml\", line 40, characters 8-38
Re-raised at file \"src/kernel/mocaml.ml\", line 48, characters 4-15
Called from file \"src/frontend/new/new_commands.ml\", line 66, characters 15-53
Called from file \"src/utils/std.ml\", line 638, characters 8-12
Re-raised at file \"src/utils/std.ml\", line 640, characters 30-39
Called from file \"src/utils/misc.ml\", line 30, characters 20-27
Re-raised at file \"src/utils/misc.ml\", line 30, characters 50-57
Called from file \"src/frontend/new/new_merlin.ml\", line 97, characters 18-54
"
```

With merlin.3.0.0 the error was different
```
cond: merlin: Error: /home/louis/.opam/frontend/lib/ocaml/map.cmt
is not a compiled interface
```

**Comment by trefis** [comment]
Apparently this bug is present on all versions from 4.02 to 4.05 (both included), and absent from 4.06 and 4.07.

**Comment by trefis** [comment]
Turns out, the actual bug was present in every version, but starting with 4.06 `.cmi` files started storing locations by default, which we used as fallback and so the error was never apparent.

The `assert false` shouldn't happen anymore.
